db/external_sst_file_ingestion_job.cc

```diff
@@ -792,18 +792,20 @@ Status ExternalSstFileIngestionJob::GetIngestedFileInfo(
     }
     file_to_ingest->smallest_internal_key.SetFrom(key);
 
-    iter->SeekToLast();
-    auto largest = iter->key();
+    Slice largest;
     if (strcmp(cfd_->ioptions()->table_factory->Name(), "PlainTable") == 0) {
       // Not Support SeekToLast , have to iter all key
       // mostly would happen in PlainTable
       iter->SeekToFirst();
-
+      largest = iter->key();
       for (; iter->Valid(); iter->Next()) {
         if (cfd_->internal_comparator().Compare(iter->key(), largest) > 0) {
           largest = iter->key();
         }
       }
```

Comment:
Also need to check iter->status() after iter->Valid() returns false, and return early if it's not ok status.

Reply:
Got it. Thanks.

Reply:
Now every time call iter->key(), it will do a status check.

Reply:
status() only needs to be checked if Valid() returns false. I added a small fix for that.